### PR TITLE
Don't include "min" suffix in build JavaScript when targeting core.

### LINKED ISSUE #26994: Different languages on the dashboard
Woocommerce 4.3 is set to German, but on my dashboard texts are still displayed in english:

woocommerce_4_3_dashboard

The translation files are correct, apparently there is an error in the output.

Many thanks in advance!

Best regards!

### PR DESCRIPTION
**Pull request by jeffstieler**
Files with "min" are ignored by GlotPress / translate.wp.org.

Fixes https://github.com/woocommerce/woocommerce/issues/26994.

This PR seeks to remove "min" from the minified JS files packaged up for the core build. This is to ensure that the files are parsed by GlotPress and language pack files are built.

### Detailed test instructions:

Testing this is definitely tricky. There really isn't a way to test this in situ where it will actually solve the problem.

- Run a normal build, e.g. `npm start` or `npm run build`
- Verify that `*.js` and `*.min.js` files are built in `/dist`
- Verify that loading script files works with and without `SCRIPT_DEBUG` defined as `true`
- Delete the `/dist` dir
- Run a core build: `WC_ADMIN_PHASE=core npm run build`
- Verify there are no `*.min.js` files in `/dist`
- Verify the JS contents are minified
- Verify that loading script files works with and without `SCRIPT_DEBUG` defined as `true`

### Changelog Note:

Fix: ensure GlotPress finds JavaScript files when building language packs for core WooCommerce.

### CODE REVIEW

**Comment by samueljseay** [comment]
> Verify that loading script files works with and without SCRIPT_DEBUG defined as true

@jeffstieler just want to walk through some of my assumptions here to make sure I understand correctly.

When we package the wc-admin package to be bundled with wc core:

1. we remove the min suffix but the files are still minified for the build?
2. since there are no more `min` suffixed files, the UnminifyWebpackPlugin now has no effect?
3. `SCRIPT_DEBUG` will basically have no effect now, we will always serve the file without a `min` suffix, but its a bit misleading to a user in this environment because the file *will* in fact be minified?

Once wc-core is built will they then minify our files?

File: config/core.json
Comment (psealock) [inline config/core.json:14]: Is is fair to say we're doing this as a temporary measure? If not, I'd find this confusing and expect that our core production release would have `minified-js: true`. But, maybe I'm thinking about this wrong.

**Review by psealock — COMMENTED** [review]
This tests well according to instructions. No issues found. I'll let Sam have final approval.

**Comment by jeffstieler** [comment]
> When we package the wc-admin package to be bundled with wc core:
> 
> 1. we remove the min suffix but the files are still minified for the build?

Correct. Core wants to keep the build small, so we are still minifying the file contents. GlotPress (translate.wordpress.org) will ignore files with the `.min` suffix, so we omit that in the core build.

> 2. `SCRIPT_DEBUG` will basically have no effect now, we will always serve the file without a `min` suffix, but its a bit misleading to a user in this environment because the file will in fact be minified?

Yes, when loading WooCommerce Admin from the core package, `SCRIPT_DEBUG` has no effect. This is true before introducing this change - just that `.min` was in the filename.

File: config/core.json
Comment (jeffstieler) [reply config/core.json:14]: This feature flag only supports the lack of both minified and unminified JS files in the core build. I chose to rename it because by outward appearance, there are no `*.min.js` files (even though the contents are actually minified).

If we were able to include unminified JS or soucemaps, etc, without inflating the core build size - we might be able to remove this temporary measure.

**Review by octaedro — APPROVED** [review]
Nice solution @jeffstieler. This is testing well here and code LGTM :shipit:

**Pull request merged**
